CMake/configcheck/final.cxx

```diff
@@ -5,21 +5,22 @@ class foo final
 
 struct Base
 {
-    virtual void foo();
+  virtual ~Base();
+  virtual void foo();
 };
 
 struct A : Base
 {
-    virtual void foo() final; // A::foo is final
-    void bar() final; // Error: non-virtual function cannot be final
+  virtual void foo() final; // A::foo is final
+//   void bar() final; // Error: non-virtual function cannot be final
```

Comment:
Can you explain why this is commented out?  Or maybe the question is: what was this doing before?

Reply:
This line is, per the comment, a compile error on compilers that support `final`. Thus, it made the test fail to compile, resulting in an incorrect determination that the compiler does *not* support `final`. Same for the other commented-out line.